src/main/java/io/supertokens/signingkeys/SigningKeys.java

```diff
@@ -194,6 +194,19 @@ public KeyInfo getLatestIssuedDynamicKey()
         }
     }
 
+    public long getCacheDurationInSeconds()
+            throws StorageQueryException, StorageTransactionLogicException, TenantOrAppNotFoundException,
+            UnsupportedJWTSigningAlgorithmException {
+        CoreConfig config = Config.getConfig(appIdentifier.getAsPublicTenantIdentifier(), main);
+
+        long timeLeftForNewKeyCreation = (
+                getLatestIssuedDynamicKey().createdAtTime
+                        + config.getAccessTokenDynamicSigningKeyUpdateInterval()
+                        - AccessTokenSigningKey.getInstance(appIdentifier, main).getDynamicSigningKeyOverlapMS()
+                        - System.currentTimeMillis()) / 1000;
+        return Math.max(timeLeftForNewKeyCreation, 0);
```

Comment:
we may want to set some minimum (i.e.: 1sec or `getDynamicSigningKeyOverlapMS`) instead of 0. Not sure, but please consider it.